src/python.rs

```diff
@@ -72,6 +103,76 @@ impl Record {
         }
         Ok(())
     }
+
+    #[new]
+    #[pyo3(signature = (id, seq, qual=None))]
+    fn new(id: String, seq: String, qual: Option<String>) -> PyResult<Record> {
+        // If `qual` is not None, check if it has the same length as `seq`
+        if let Some(qual) = &qual {
+            if qual.len() != seq.len() {
+                return Err(PyValueError::new_err(
+                    "Sequence and quality strings must have the same length",
+                ));
+            }
+        }
+        Ok(Record { id, seq, qual })
+    }
+
+    pub fn __hash__(&self) -> PyResult<u64> {
+        let mut hasher = DefaultHasher::new();
+        self.id.hash(&mut hasher);
+        self.seq.hash(&mut hasher);
+        match &self.qual {
+            Some(qual) => qual.hash(&mut hasher),
+            None => {}
+        }
+        Ok(hasher.finish())
+    }
+
+    pub fn __eq__(&self, other: &Record) -> PyResult<bool> {
+        Ok(self.id == other.id && self.seq == other.seq && self.qual == other.qual)
+    }
+
+    pub fn __len__(&self) -> PyResult<usize> {
+        Ok(self.seq.len())
+    }
+
+    pub fn __str__(&self) -> PyResult<String> {
+        if self.qual.is_none() {
+            let wrapped_seq = self
+                .seq
+                .as_bytes()
+                .chunks(60)
+                .map(|chunk| String::from_utf8_lossy(chunk).to_string())
+                .collect::<Vec<String>>()
+                .join("\n");
+            Ok(format!(">{}\n{}", self.id, wrapped_seq))
```

Comment:
I'm on the fence about whether these should end in a newline. BioPython does add a newline and I'm apt to follow convention. Maybe a tie breaker is that a `\n` denotes a record separator while reading a FASTA so it should be present when writing a FASTA record.